From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Philippe=20Tr=C3=A9buchet?= <clipos@ssi.gouv.fr>
Date: Thu, 28 Jul 2016 19:24:01 +0200
Subject: [PATCH 05/32] clip: vserver/grsec: fs/proc: Fix access to
 task->nsproxy

Upstream change impacting 1504_vserver-nsproxy-and-leak-fixes-1.patch .

Patch: 1504_vserver-nsproxy-and-leak-fixes-2.patch
---
 fs/proc/array.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/fs/proc/array.c b/fs/proc/array.c
index 283043fa4864b..e273c178cc3c0 100644
--- a/fs/proc/array.c
+++ b/fs/proc/array.c
@@ -392,8 +392,8 @@ int proc_pid_nsproxy(struct seq_file *m, struct pid_namespace *ns,
 {
 	struct nsproxy *nsproxy;
 
-	rcu_read_lock();
-	nsproxy = task_nsproxy(task);
+	task_lock(task);
+	nsproxy = task->nsproxy;
 	if (nsproxy == NULL)
 		goto out;
 
@@ -420,7 +420,7 @@ int proc_pid_nsproxy(struct seq_file *m, struct pid_namespace *ns,
 			(nsproxy->net_ns == init_task.nsproxy->net_ns ? 'I' : '-'));
 
 out:
-	rcu_read_unlock();
+	task_unlock(task);
 	return 0;
 }
 
-- 
2.17.0

